### DISCUSSION TITLE
Keycloak Email & usernameauth

### DISCUSSION THREAD

**Discussion by ShakoG2** [keycloak/keycloak#33192] [Q&A]
Hi , I use keycloak version 25 . I want to make something like this . when user register with google identity provider and also make registration with classic way I want to create 2 user this user I want to turn on email duplicates option from realm configuration , becouse when user make registration with classic way it throw error email already exist becouse google registaration already set this email before. when I turn on duplicate emails now I cant turn on "email as a username" option or "loginWithEmail". the problem is that I want to login with email and also with username. and I can't do it. can someone help me ?

**Comment by codespearhead** [comment]
Very few platforms allow that because it leaves room for massive security issues, unless you can fully trust the external identity provider, such as logging into your Instagram account from your Facebook (they're owned by the same company) or you can use Norway's BankID (highly trusted authentication provider) to access most public Norwegian websites.

Either way, you can look into the source of the following quote and implement it.

> The AutoLink authenticator is dangerous in a generic environment where users can register themselves using arbitrary usernames or email addresses. Do not use this authenticator unless you are carefully curating user registration and assigning usernames and email addresses. [[Source](https://www.keycloak.org/docs/latest/server_admin/index.html#automatically-link-existing-first-login-flow)]
